syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";
import "google/protobuf/wrappers.proto";
import "user_message.proto";
package doodle;




message metadata_database{
  uint64 id = 1;
  string uuid_path = 2;
  google.protobuf.UInt64Value parent = 3;
  google.protobuf.Timestamp update_time = 4;
  google.protobuf.Any Metadata_cereal = 5;

  message meta_type_value{
    uint64 value = 1;
  }
  meta_type_value m_type = 6;
}


message metadata_database_filter{
  uint64 id = 1;
  google.protobuf.Timestamp begin_time = 2;
  google.protobuf.Timestamp end_time = 3;
  metadata_database.meta_type_value m_type = 4;
  google.protobuf.UInt64Value parent = 5;
}

service metadata_server{
  rpc install_metadata(metadata_database) returns(metadata_database) {}
  rpc delete_metadata(metadata_database) returns(metadata_database) {}
  rpc update_metadata(metadata_database) returns(metadata_database) {}

  rpc filter_metadata(metadata_database_filter) returns(stream metadata_database) {}

  rpc install_user_date(user_database) returns(user_database) {}
  rpc update_user_date(user_database) returns(user_database) {}
  rpc delete_user_date(user_database_filter) returns(user_database) {}

  rpc filter_user_date(user_database_filter) returns(stream user_database) {}


}
